util/zenfs.cc

```diff
@@ -69,21 +70,56 @@ Status zenfs_mount(std::unique_ptr<ZonedBlockDevice> &zbd,
   return s;
 }
 
+// Create or check pre-existing aux directory and fail if it is
+// inaccessible by current user and if it has previous data
+int create_aux_path() {
+  struct dirent *dent;
+  DIR *aux_dir;
+  unsigned int nfiles = 0;
+  int ret = 0;
+
+  errno = 0;
+  ret = mkdir(FLAGS_aux_path.c_str(), S_IRWXU | S_IRWXG | S_IROTH | S_IXOTH);
+  if (ret < 0 && EEXIST != errno) {
+    fprintf(stderr, "ZenFS: Error creating directory %s: %s\n",
+            FLAGS_aux_path.c_str(), strerror(errno));
+    return 1;
+  }
+  // The aux_path is now available, check if it is not empty and
+  // the user has access permission
+
+  errno = 0;
+  aux_dir = opendir(FLAGS_aux_path.c_str());
+  if (errno) {
+    fprintf(stderr, "ZenFS: Error opening directory %s: %s\n",
+            FLAGS_aux_path.c_str(), strerror(errno));
+    return 1;
+  }
+
+  while ((dent = readdir(aux_dir)) != NULL && nfiles <= 2) nfiles++;
+  if (nfiles > 2) {
```

Comment:
Please add a comment on this condition, why > 2?

Reply:
Another corner case is reading the content of the `/` (root directory) - in this case we won't have `.` and `..` in the output.
Although I must admit that specifying `aux_dir=/` looks unrealistic to me (may only be meaningful under something like `chroot`).

Reply:
Added the comment.
I can see  . and .. when i do a "ls -a /" , and anyway mkfs will fail when run on /, unless it has less than 2 files, in which case it is already in a bad shape.

Reply:
OK, then